types/duration.pp

```diff
@@ -0,0 +1 @@
+type Nginx::Duration = Pattern[/^((\d+y)?\s?(\d+M)?\s?(\d+w)?\s?(\d+d)?\s?(\d+h)?\s?(\d+m)?\s?(\d+s)?\s?(\d+ms)?\s?|\d+)$/]
```

Comment:
why is there a trailing `\s?`?

Reply:
diffing this regex with the stricter pattern that's `Nginx::Time` is still on my todo list.

Reply:
I added a commit that merges it into Nginx::Time. That also copies the Nginx docs. While it now allows a lot of additional whitespace, it's probably easier to read this way. The various suffixes can be combined and I can't see how the regex can be really improved.